.github/workflows/release.yml

```diff
@@ -0,0 +1,81 @@
+name: 'Release'
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+env:
+  CARGO_INCREMENTAL: 0
+
+jobs:
+  build:
+    name: Binary
+    strategy:
+      fail-fast: false
+      matrix:
+        job:
+          - { target: x86_64-unknown-linux-musl, exe: amd64-linux, os: ubuntu-latest }
+          - { target: aarch64-unknown-linux-musl, exe: aarch64-linux,  os: ubuntu-latest }
+          - { target: armv7-unknown-linux-musleabi, exe: armv7-linux, os: ubuntu-latest }
+          - { target: wasm32-wasi, exe: wasi.wasm, os: ubuntu-latest }
+          - { target: x86_64-apple-darwin, exe: macos, os: macos-latest }
+          - { target: x86_64-pc-windows-msvc, exe: windows.exe, os: windows-2019 }
+    runs-on: ${{ matrix.job.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: 1.62.0
+          override: true
+          target: ${{ matrix.job.target }}
+          components: rust-src
+      - uses: actions-rs/cargo@v1
+        with:
+          use-cross: true
+          args: --release --target=${{ matrix.job.target }} --locked
+          command: build
+      - name: Rename result
+        run: |
+          rm target/${{ matrix.job.target }}/release/wasm-custom-section.d
+          cp target/${{ matrix.job.target }}/release/wasm-custom-section* wasm-custom-section-${{ matrix.job.exe }}
+      - name: Archive production artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: arty
+          path: wasm-custom-section-${{ matrix.job.exe }}
+
+  wapm-publish:
+    runs-on: ubuntu-20.04
+    needs: build
+    steps:
+      - name: Setup Wasmer
+        uses: wasmerio/setup-wasmer@v1
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v3
+        with:
+          name: arty
+      - run: mkdir -p target/wasm32-wasi/release/ && mv wasm-custom-section-wasi.wasm target/wasm32-wasi/release/wasm-custom-section.wasm
+      - name: WAPM publish
+        run: wapm publish
+        env:
+          WAPM_REGISTRY_TOKEN: ${{ secrets.WAPM_REGISTRY_TOKEN }}
```

Comment:
If you're not interested in the wapm.io thing, I can get rid of this part.

Otherwise, you'd have to get an account there, create a new token, and add it as a repository secret.
![image](https://user-images.githubusercontent.com/1753388/179006488-b313cb2f-3394-4790-976d-51263eac1d5d.png)
![image](https://user-images.githubusercontent.com/1753388/179006323-4773b4e8-e1dd-4790-b13e-7330c52480b5.png)